test/linters_test.go

```diff
@@ -252,11 +252,55 @@ func TestExtractRunContextFromComments(t *testing.T) {
 	assert.Equal(t, []string{"-Egoimports"}, rc.args)
 }
 
-func TestGolintConsumesXTestFiles(t *testing.T) {
-	dir := getTestDataDir("withxtest")
-	const expIssue = "`if` block ends with a `return` statement, so drop this `else` and outdent its block"
+func TestTparallel(t *testing.T) {
+	t.Run("should fail on missing top-level Parallel()", func(t *testing.T) {
+		sourcePath := filepath.Join(testdataDir, "tparallel", "missing_toplevel_test.go")
+		args := []string{
+			"--disable-all", "--print-issued-lines=false", "--print-linter-name=false", "--out-format=line-number", "--enable", "tparallel",
+			sourcePath,
+		}
+		rc := extractRunContextFromComments(t, sourcePath)
+		args = append(args, rc.args...)
+
+		cfg, err := yaml.Marshal(rc.config)
+		assert.NoError(t, err)
+
+		testshared.NewLintRunner(t).RunWithYamlConfig(string(cfg), args...).
+			ExpectHasIssue(
+				"testdata/tparallel/missing_toplevel_test.go:7:6: TestTopLevel should call t.Parallel on the top level as well as its subtests\n",
+			)
+	})
+
+	t.Run("should fail on missing subtest Parallel()", func(t *testing.T) {
+		sourcePath := filepath.Join(testdataDir, "tparallel", "missing_subtest_test.go")
+		args := []string{
+			"--disable-all", "--print-issued-lines=false", "--print-linter-name=false", "--out-format=line-number", "--enable", "tparallel",
+			sourcePath,
+		}
+		rc := extractRunContextFromComments(t, sourcePath)
+		args = append(args, rc.args...)
+
+		cfg, err := yaml.Marshal(rc.config)
+		assert.NoError(t, err)
+
+		testshared.NewLintRunner(t).RunWithYamlConfig(string(cfg), args...).
+			ExpectHasIssue(
+				"testdata/tparallel/missing_subtest_test.go:7:6: TestSubtests's subtests should call t.Parallel\n",
+			)
+	})
+
+	t.Run("should pass on parallel test with no subtests", func(t *testing.T) {
```

Comment:
This test will fail because of a false-positive. See https://github.com/moricho/tparallel/issues/8

Reply:
I fixed that bug and released a new version.
https://github.com/moricho/tparallel/releases/tag/v0.2.1

Reply:
Awesome, thanks